libraries/hyperhtml/src/component-tests.js

```diff
@@ -0,0 +1,222 @@
+  app.innerHTML = '';
+  root = null;
+});
+
+describe('no children', function() {
+  it('can display a Custom Element with no children', function() {
+    root.appendChild(ComponentWithoutChildren());
+    let wc = root.querySelector('#wc');
+    expect(wc).toExist();
+  });
+});
+
+describe('with children', function() {
+  function expectHasChildren(wc) {
+    expect(wc).toExist();
+    let shadowRoot = wc.shadowRoot;
+    let heading = shadowRoot.querySelector('h1');
+    expect(heading).toExist();
+    expect(heading.textContent).toEqual('Test h1');
+    let paragraph = shadowRoot.querySelector('p');
+    expect(paragraph).toExist();
+    expect(paragraph.textContent).toEqual('Test p');
+  }
+
+  it('can display a Custom Element with children in a Shadow Root', function() {
+    root.appendChild(ComponentWithChildren());
+    let wc = root.querySelector('#wc');
+    expectHasChildren(wc);
+  });
+
+  it('can display a Custom Element with children in a Shadow Root and pass in Light DOM children', function(done) {
+    root.appendChild(new ComponentWithChildrenRerender());
+    setTimeout(function () {
+      let wc = root.querySelector('#wc');
+      expectHasChildren(wc);
+      expect(wc.textContent.includes('2')).toEqual(true);
+      done();
+    }, 10);
+  });
+
+  it('can display a Custom Element with children in the Shadow DOM and handle hiding and showing the element', function(done) {
+    let ce = root.appendChild(new ComponentWithDifferentViews);
+    setTimeout(function () {
+      let wc = root.querySelector('#wc');
+      expectHasChildren(wc);
+      ce.toggle();
+      let dummy = root.querySelector('#dummy');
+      expect(dummy).toExist();
+      expect(dummy.textContent).toEqual('Dummy view');
+      ce.toggle();
+      expect(wc).toBe(root.querySelector('#wc'));
+      expectHasChildren(wc);
+      ce.toggle();
+      expect(dummy).toBe(root.querySelector('#dummy'));
+      done();
+    }, 10);
+  });
+
+});
+
+describe('attributes and properties', function() {
+  it('will pass boolean data as either an attribute or a property', function() {
+    root.appendChild(ComponentWithProperties());
+    let wc = root.querySelector('#wc');
+    let data = wc.bool || wc.hasAttribute('bool');
+    expect(data).toBe(true);
+  });
+
+  it('will pass numeric data as either an attribute or a property', function() {
+    root.appendChild(ComponentWithProperties());
+    let wc = root.querySelector('#wc');
+    let data = wc.num || wc.getAttribute('num');
+    expect(data).toEqual(42);
+  });
+
+  it('will pass string data as either an attribute or a property', function() {
+    root.appendChild(ComponentWithProperties());
+    let wc = root.querySelector('#wc');
+    let data = wc.str || wc.getAttribute('str');
+    expect(data).toEqual('hyperHTML');
+  });
+
+  it('will pass array data as a property', function() {
+    root.appendChild(ComponentWithProperties());
+    let wc = root.querySelector('#wc');
+    let data = wc.data.arr;
```

Comment:
It looks like this component has created a new property on the element called `data`. Is that correct? To fulfill the test it needs to set the `arr` property that's already defined https://github.com/webcomponents/custom-elements-everywhere/blob/master/libraries/__shared__/webcomponents/src/ce-with-properties.js#L37